### PR TITLE
Try unbreaking msys2

### PR DESCRIPTION
**Pull request by cocreature**
yandex has apparently desided to return 443s.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by SamirTalwar-DA — APPROVED** [review]
:crossed_fingers:

**Comment by SamirTalwar-DA** [comment]
I guess not.

```
The remote server returned an error: (404) Not Found.
URL https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.c-1-any.pkg.tar.xz#/openssl.msys2 is not valid
```

File: dev-env/windows/manifests/msys2.json [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
-        "https://mirror.yandex.ru/mirrors/msys2/mingw/x86_64/mingw-w64-x86_64-jq-1.6-2-any.pkg.tar.xz#/jq.msys2",
-        "https://mirror.yandex.ru/mirrors/msys2/msys/x86_64/gnu-netcat-0.7.1-1-x86_64.pkg.tar.xz#/netcat.msys2",
-        "https://mirror.yandex.ru/mirrors/msys2/msys/x86_64/patch-2.7.6-1-x86_64.pkg.tar.xz#/patch.msys2",
-        "https://mirror.yandex.ru/mirrors/msys2/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.c-1-any.pkg.tar.xz#/openssl.msys2",
-        "https://mirror.yandex.ru/mirrors/msys2/mingw/x86_64/mingw-w64-x86_64-postgresql-11.3-1-any.pkg.tar.xz#/pgsql.msys2"
+        "https://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20180531.tar.xz",
+        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-jq-1.6-2-any.pkg.tar.xz#/jq.msys2",
+        "https://repo.msys2.org/msys/x86_64/gnu-netcat-0.7.1-1-x86_64.pkg.tar.xz#/netcat.msys2",
+        "https://repo.msys2.org/msys/x86_64/patch-2.7.6-1-x86_64.pkg.tar.xz#/patch.msys2",
+        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.c-1-any.pkg.tar.xz#/openssl.msys2",
```
Comment (SamirTalwar-DA) [inline dev-env/windows/manifests/msys2.json:9]: Looks like our OpenSSL version is too old for the main repo.

```suggestion
        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-openssl-1.1.1.g-1-any.pkg.tar.xz#/openssl.msys2",
```

There's a newer one but it's got a `zst` extension and I don't know if dev-env supports that.

Comment (cocreature) [reply dev-env/windows/manifests/msys2.json:9]: I think it’s more complex than that. This still gives you a tarball not a `.msys2` file.

Comment (cocreature) [reply dev-env/windows/manifests/msys2.json]: trying anyway because :shrug:

File: dev-env/windows/manifests/msys2.json [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
+        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-postgresql-11.3-1-any.pkg.tar.xz#/pgsql.msys2"
```
Comment (SamirTalwar-DA) [inline dev-env/windows/manifests/msys2.json:10]: This one will also break.

```suggestion
        "https://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-postgresql-12.2-1-any.pkg.tar.xz#/pgsql.msys2"
```

File: dev-env/windows/manifests/msys2.json (hunk 1 above)
Comment (SamirTalwar-DA) [reply dev-env/windows/manifests/msys2.json]: Oh, hash failure. I forgot we are Highly Secure™. :slightly_frowning_face:

Comment (cocreature) [reply dev-env/windows/manifests/msys2.json]: security is measured in the number of ™s that you include.

Comment (cocreature) [reply dev-env/windows/manifests/msys2.json]: ah the `#/` stuff is apparently scoop magic so it doesn’t matter.

**Review by SamirTalwar-DA — APPROVED** [review]
Ship it!

**Pull request merged**